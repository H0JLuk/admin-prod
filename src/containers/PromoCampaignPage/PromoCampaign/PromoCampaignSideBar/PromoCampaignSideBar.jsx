import React from 'react';
import cn from 'classnames';

import sberLogo from '../../../../static/images/sber-logo.png';

import styles from './PromoCampaignSideBar.module.css';

const NEW_PROMO_CAMPAIGN_STEPS = ['Новая промо-кампания', 'Добавление контента', 'Настройки видимости'];
const PROMO_CAMPAIGN_STEPS = ['Промо-кампания', 'Контент промо-кампании', 'Настройки видимости'];

const IMG_ALT = 'sberLogo';

const PromoCampaignSideBar = ({ active, onClick, validStep, newPromoCampaign, hideLastStep }) => {
    const currentSteps = newPromoCampaign
        ? NEW_PROMO_CAMPAIGN_STEPS
        : !hideLastStep ? PROMO_CAMPAIGN_STEPS : PROMO_CAMPAIGN_STEPS.slice(0, PROMO_CAMPAIGN_STEPS.length - 1);

    return (
        <div className={ styles.menu }>
            <div className={ styles.blockLogo }>
                <img className={ styles.logo } src={ sberLogo } alt={ IMG_ALT } />
            </div>
            { currentSteps.map((step, index) => (
                <MenuItem
                    key={ step }
                    disabled={ validStep < index + 1 }
                    onClick={ onClick }
                    title={ step }
                    active={ index === active - 1 }
                    index={ index }
                />
            )) }
        </div>
        );
};

export default PromoCampaignSideBar;


const MenuItem = ({ title, active, onClick, disabled, index }) => {
    const onClickHandler = () => {
        if (!disabled) {
            onClick(index + 1);
        }
    };

    return (
        <div
            className={ cn(
                styles.menuItem,
                {
                    [styles.active]: active,
                    [styles.disabled]: disabled,
                })
            }
            onClick={ onClickHandler }
        >
            { title }
        </div>
    );
};
