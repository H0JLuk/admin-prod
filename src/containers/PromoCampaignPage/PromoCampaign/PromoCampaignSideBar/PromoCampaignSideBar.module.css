.menu {
    position: sticky;
    top: 0;
    width: 300px;
    max-height: 100vh;
    min-height: 100vh;
    flex-shrink: 0;
    background-color: white;
    border-right: 1px solid #d4d9de;
}

.blockLogo {
    border-bottom: 1px solid #d4d9de;
    margin-bottom: 15px;
}

.logo {
    width: 45%;
    padding: 17px;
}

.menuItem {
    display: flex;
    text-decoration: none;
    font-size: 12px;
    font-weight: 600;
    line-height: 1.5;
    letter-spacing: -0.3px;
    background-color: transparent;
    padding: 10px 41px;
    cursor: pointer;
    position: relative;
}

.menuItem::before {
    content: '';
    position: absolute;
    left: 24px;
    bottom: 0;
    width: 1.1px;
    height: 38px;
    background-color: #dedede;
}

.menuItem:hover {
    background-color: #f8f8f8;
}

.active {
    color: #09a552 !important;
    position: relative;
}

.active::before {
    background-color: #09a552;
}

.disabled {
    color: #858585 !important;
    background-color: #f8f8f8;
    cursor: not-allowed;
}
