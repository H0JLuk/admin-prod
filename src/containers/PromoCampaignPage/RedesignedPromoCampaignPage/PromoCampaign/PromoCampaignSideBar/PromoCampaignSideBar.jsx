import React, { useCallback } from 'react';
import cn from 'classnames';

import sberLogo from '../../../../../static/images/sber-logo.png';

import styles from './PromoCampaignSideBar.module.css';

const NEW_PROMO_CAMPAIGN_STEPS = ['Новая промо-кампания', 'Добавление контента', 'Настройки видимости'];
const PROMO_CAMPAIGN_STEPS = ['Промо-кампания', 'Контент промо-кампании', 'Настройки видимости'];

const IMG_ALT = 'sberLogo';

const PromoCampaignSideBar = ({ active, onClick, validStep, newPromoCampaign }) => (
    <div className={ styles.menu }>
        <div className={ styles.blockLogo }>
            <img className={ styles.logo } src={ sberLogo } alt={ IMG_ALT } />
        </div>
        { (newPromoCampaign ? NEW_PROMO_CAMPAIGN_STEPS : PROMO_CAMPAIGN_STEPS).map((step, index) => (
            <MenuItem
                key={ step }
                disabled={ validStep < index + 1 }
                onClick={ onClick }
                title={ step }
                active={ index === active - 1 }
                index={ index }
            />
        ))}
    </div>
);

export default PromoCampaignSideBar;


const MenuItem = ({ title, active, onClick, disabled, index }) => {
    const onClickHandler = useCallback(() => {
        if (!disabled) {
            onClick(index + 1);
        }
    }, [onClick, index, disabled]);

    return (
        <div
            className={ cn(
                styles.menuItem,
                {
                    [styles.active]: active,
                    [styles.disabled]: disabled,
                })
            }
            onClick={ onClickHandler }
        >
            { title }
        </div>
    );
};
