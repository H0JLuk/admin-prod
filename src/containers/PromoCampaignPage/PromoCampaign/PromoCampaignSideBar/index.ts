import PromoCampaignSideBar from './PromoCampaignSideBar';

export default PromoCampaignSideBar;
