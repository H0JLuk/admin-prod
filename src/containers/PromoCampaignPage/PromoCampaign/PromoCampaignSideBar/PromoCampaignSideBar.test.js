import React from 'react';
import { shallow, mount } from 'enzyme';
import PromoCampaignSideBar from './PromoCampaignSideBar';

describe('<PromoCampaignSideBar /> test', () => {
    const initialProps = {
        active: 1,
        onClick: jest.fn(),
        validStep: 2,
        newPromoCampaign: true,
        hideLastStep: false,
    };

    it('should match the snapshot', () => {
        const container = shallow(<PromoCampaignSideBar { ...initialProps } />);
        expect(container.html()).toMatchSnapshot();
    });

    it('if promo campaign is new, currentSteps length should be 3', () => {
        const container = shallow(<PromoCampaignSideBar { ...initialProps } />);
        expect(container.find('MenuItem')).toHaveLength(3);
    });

    it('if promo campaign is not new, currentSteps length should be 3', () => {
        const container = shallow(<PromoCampaignSideBar { ...{ ...initialProps, newPromoCampaign: false } } />);
        expect(container.find('MenuItem')).toHaveLength(3);
    });

    it('if promo campaign is not new and the last step is hidden, currentSteps length should be 2', () => {
        const container = shallow(<PromoCampaignSideBar { ...{ ...initialProps, newPromoCampaign: false, hideLastStep: true } } />);
        expect(container.find('MenuItem')).toHaveLength(2);
    });

    it('callback should be called', () => {
        const container = mount(<PromoCampaignSideBar { ...initialProps } />);
        container.find('MenuItem').find('div').at(0).simulate('click');
        expect(initialProps.onClick).toBeCalledTimes(1);
        expect(initialProps.onClick).toBeCalledWith(1);
    });

    it('callback should not be called', () => {
        const container = mount(<PromoCampaignSideBar { ...initialProps } />);
        container.find('MenuItem').find('div').at(2).simulate('click');
        expect(initialProps.onClick).not.toBeCalled();
    });
});
