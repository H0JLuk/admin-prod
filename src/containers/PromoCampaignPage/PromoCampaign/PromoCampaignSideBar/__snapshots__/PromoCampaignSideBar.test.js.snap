// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PromoCampaignSideBar /> test should match the snapshot 1`] = `"<div class=\\"menu\\"><div class=\\"blockLogo\\"><img class=\\"logo\\" src=\\"sber-logo.png\\" alt=\\"sberLogo\\"/></div><div class=\\"menuItem active\\">Новая промо-кампания</div><div class=\\"menuItem\\">Добавление контента</div><div class=\\"menuItem disabled\\">Настройки видимости</div></div>"`;
